pyproject.toml

```diff
@@ -1,36 +1,50 @@
-[tool.poetry]
-name = "nzshm-model-graphql-api"
-version = "0.3.1"
-description = ""
-authors = ["NSHM <nshm@gns.cri.nz>", "Chris Chamberlain <chrisbc@artisan.co.nz>"]
-license = "AGPL3"
+[project]
+name = "nshm-model-graphql-api"
+version = "0.4.0"
+description = "A graphql API for nzshm-model library."
+authors = [
+    {name = "Chris B Chamberlain",email = "chrisbc@artisan.co.nz"},
+    {name="NSHM Compute Working Group", email="nshm@gns.cri.nz"}
+]
 readme = "README.md"
+requires-python = "^3.12"
+license = "AGPL-3.0-or-later"
 packages = [{include = "nshm_model_graphql_api"}]
 
-[tool.poetry.dependencies]
-python = "^3.10"
-nzshm-model = "0.10.6"
-flask = "^3.0.3"
-flask-cors = "^4.0.1"
-pyyaml = "^6.0.1"
-graphql-server = {extras = ["flask"], version = "3.0.0b7"}
-graphene = "^3.3"
-# nzshm-model = "^0.9.3"
+dependencies = [
+    "boto3 (>=1.24.57)",
+    "flask (>=3.0.3)",
+    "flask-cors (>=6.0)",
+    "graphene (>=3.3)",
+    "graphql-server (==3.0.0b7)",
+    "python-dateutil (>=2.8.2)",
+    "pyyaml (>=6.0.1)",
+    "nzshm-model (>=0.14.0)",
+]
 
-[tool.poetry.group.dev.dependencies]
-tox = "^4.2.8"
-black  = { version = "^24.3.0"}
-isort  = { version = "^5.8.0"}
-flake8  = { version = "^3.9.2"}
-flake8-docstrings = { version = "^1.6.0", optional = true }
-pytest = "^7.4.0"
-pytest-cov  = { version = "^2.12.0"}
-virtualenv  = { version = "^20.2.2", optional = true}
-pip  = { version = "^23.2.1", optional = true}
-bump2version = {version = "^1.0.1"}
-mypy = "^1.5.1"
-types-pyyaml = "^6.0.12.20240311"
+[tool.poetry.group.dev]
+optional = true
+
+[dependency-groups]
+dev = [
+    "black",
+    "bump2version",
+    "flake8-docstrings",
+    "flake8",
+    "isort",
+    "moto (>=5.1)",
+    "mypy",
+    "pip-audit",
+    "pre-commit",
+    "pytest-cov",
+    "pytest",
+    "safety",
+    "sgqlc",
+    "tox",
+    "types-python-dateutil",
+    "types-pyyaml"
+]
 
 [build-system]
-requires = ["poetry-core"]
+requires = ["poetry-core>=2.2"]
```

Comment:
we have been using poetry-core>=2.2,<3.0 in other repos. Any reason to restrict the upper bound?

Reply:
let's stick with <3 until that version is released/tested.